server/src/main/java/org/elasticsearch/search/vectors/KnnVectorQueryBuilder.java

```diff
@@ -199,11 +212,14 @@ public String getWriteableName() {
     }
 
     @Override
-    protected QueryBuilder doRewrite(QueryRewriteContext queryRewriteContext) throws IOException {
+    protected QueryBuilder doRewrite(QueryRewriteContext ctx) throws IOException {
         boolean changed = false;
+        if (ctx instanceof SearchExecutionContext searchExecutionContext && searchExecutionContext.getAliasFilter() != null) {
+            filterQueries.add(searchExecutionContext.getAliasFilter());
+        }
```

Comment:
How does this rewrite ever exit and do we continually append the alias filter on subsequent rewrites?

What do you think about adding this code to `doToQuery` instead of rewrite? It seems to me that the aliasFilter will be rewritten already and then when we call `doToQuery` we take the filter and append it to the list (not updating the query object, but instead adjusting the boolean query directly)

Like:
```
        BooleanQuery.Builder builder = new BooleanQuery.Builder();
        for (QueryBuilder query : this.filterQueries) {
            builder.add(query.toQuery(context), BooleanClause.Occur.FILTER);
        }
        if (searchExecutionContext.getAliasFilter() != null) {
            builder.add(searchExecutionContext.getAliasFilter().toQuery(context), BooleanClause.Occur.FILTER);
        }
        BooleanQuery booleanQuery = builder.build();
        Query filterQuery = booleanQuery.clauses().isEmpty() ? null : booleanQuery;

```

Reply:
Ben, indeed I have not thought through about rewrite. Great suggestion about putting into `doToQuery`, I confirm that alias filter  is already rewritten by this stage.  Addressed in 8a4cb9b7922372317a7d01594ab85856e2ce317e.